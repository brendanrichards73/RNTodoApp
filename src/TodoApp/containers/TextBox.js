import React from 'react'
import { StyleSheet, TextInput, TouchableOpacity, View} from 'react-native'
import AntIcon from 'react-native-vector-icons/AntDesign'; 

const styles = StyleSheet.create({
    textInput: {
        borderWidth: 1, 
        borderColor: '#f2f2e1',
        backgroundColor: '#eaeaea', 
        height: 50, 
        flex: 1, 
        padding: 5
    },
    container: { height: 50, backgroundColor: '#eaeaea', alignItems:'center', justifyContent: 'center'}
})

const TextBox = ({value, onChangeText, placeholder = 'Type something', onPressIcon}) => {
    return <>
            <TextInput
                onChangeText={onChangeText} 
                value={value}
                placeholder={placeholder}
                style={styles.textInput}
            />
            <TouchableOpacity onPress={onPressIcon}>
                <View style={styles.container}>
                    <AntIcon name='plus' size={30} style={{ color: '#de9595', padding: 10}} />
                </View>
            </TouchableOpacity>
    </>
}

export default TextBox