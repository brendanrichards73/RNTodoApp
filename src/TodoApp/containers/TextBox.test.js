import TextBox from './TextBox'
import React from 'react'
import renderer from 'react-test-renderer'

describe('TextBox', () => {
    test('render default', () => {
        const tree = renderer.create(
            <TextBox value={'I need to cut the grass'}/>
        ).toJSON()

        expect(tree).toMatchSnapshot()
    })
})