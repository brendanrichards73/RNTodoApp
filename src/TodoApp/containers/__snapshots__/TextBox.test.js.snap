// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextBox render default 1`] = `
Array [
  <TextInput
    allowFontScaling={true}
    placeholder="Type something"
    rejectResponderTermination={true}
    style={
      Object {
        "backgroundColor": "#eaeaea",
        "borderColor": "#f2f2e1",
        "borderWidth": 1,
        "flex": 1,
        "height": 50,
        "padding": 5,
      }
    }
    underlineColorAndroid="transparent"
    value="I need to cut the grass"
  />,
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#eaeaea",
          "height": 50,
          "justifyContent": "center",
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 30,
            },
            Object {
              "color": "#de9595",
              "padding": 10,
            },
            Object {
              "fontFamily": "anticon",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        î˜§
      </Text>
    </View>
  </View>,
]
`;
